sound/soc/sof/topology.c

```diff
@@ -1272,7 +1272,8 @@ static int sof_connect_dai_widget(struct snd_soc_component *scomp,
 
 				return -EINVAL;
 			}
-			cpu_dai->capture_widget = w;
+			if (cpu_dai)
```

Comment:
I don't think this makes sense. First you know that `cpu_dai != NULL`, why check for this? Just adding the initialisation will already fix the compiler warning, no need to also add a check. Secondly, maybe a better way to fix this warning and avoid adding a superfluous initialisation would be to move the assignment below under the `if` line 1266. Similar below of course.

Reply:
@ranj063 yeah, I like @lyakh idea of moving capture_widget assignment to the for loop at l1266 . This makes the whole thing easier to follow.

Reply:
ack fixed now